### PR TITLE
Updating CONTRIBUTING.md for Adding Test.

### PR DESCRIPTION
**Pull request by marcocapano**
Separated rules about adding tests in a dedicated section.  Fixes #588 🚀 

## Checklist

- [ x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.

### CODE REVIEW

**Comment by marcocapano** [comment]
Since this is all about documentation, I didn't include check like "Are all extensions written in Swift 4?".
Should I write a Changelog for this kind of commit?

**Comment by marcocapano** [comment]
I removed the check since documentation is not in the "Types of changes" section.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -76,16 +77,9 @@ public extension SomeType {
 	- Instance methods & type methods
 	- Initializers
 	- Structs
-- All extensions should be tested.
+- All extensions should be tested. See "Adding Tests" to know more.
```
Comment (SD10) [inline CONTRIBUTING.md]: Can you link to the "adding tests" section here?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -106,6 +100,24 @@ public extension SomeType {}
 
 ---
 
+## Adding Tests
+
+Please refer to the following rules before submitting a pull request with your new tests:
```
Comment (SD10) [inline CONTRIBUTING.md]: How about "Please follow these guidelines before submitting a pull request adding your tests:" ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -106,6 +100,24 @@ public extension SomeType {}
...
+
+- Every extended SwifterSwift type should have one specific subclass of XCTestCase.
+- The subclass should be marked as final.
+- All extensions files and test files have a one to one relation.
```
Comment (SD10) [inline CONTRIBUTING.md]: "one to one relation" -> "one to one relationship"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -106,6 +100,24 @@ public extension SomeType {}
...
+   - (example: all tests for "**StringExtensions.swift**" are found in the "**StringExtensionsTests.swift**" file)
+- There should be a one to one relationship between methods/properties and their backing tests.
+- Tests should be named using the same API of the extension it backs.
+   - (example: `DateExtensions` method `isBetween` is named `testIsBetween`)
+- All test files are named based on the extensions which it tests.
```
Comment (SD10) [inline CONTRIBUTING.md]: Can you move this to be point #2?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -106,6 +100,24 @@ public extension SomeType {}
...
+   - (example: all String extensions tests are found in "**StringExtensionsTests.swift**" file)
+- SwifterSwift source files should not be added to the test target directly, but you should rather import SwifterSwift into the test target by using: @testable import SwifterSwift
+- Tests are ordered inside files in the same order as extensions. See "Adding new Extensions" to know more.
```
Comment (SD10) [inline CONTRIBUTING.md]: It wouldn't be bad to link to the `Adding new Extensions` section here either

**Comment by marcocapano** [comment]
Thanks for reviewing! I’m just getting started with this project so I tought starting from documentation can help. I’ll edit this based on your suggestion soon

**Comment by marcocapano** [comment]
@SD10 done! thanks for reviewing! 🚀

**Comment by omaralbeik** [comment]
Sorry about locking and closing 😅, I was trying Githawk's [context menu](https://github.com/GitHawkApp/ContextMenu) and got too excited, started tapping buttons like crazy 😂

**Comment by LucianoPAlmeida** [comment]
Hey @marcocapano can you just update this branch with master, so we can merge it :))

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃

**Comment by marcocapano** [comment]
Thanks for inviting me @SD10 . I can now delete my branch right?

**Comment by SD10** [comment]
Yeah @marcocapano and you no longer need to work on a fork of SwifterSwift. Contributors can work off of the main repo

**Pull request merged**